### PR TITLE
690: Save `reportId` for Outbound Messages

### PR DESCRIPTION
**Pull request by halprin**
# Save `reportId` for Outbound Messages

We have pivoted away from saving `submissionId` to saving `reportId` for outbound messages sent to RS.

## Issue

#690.

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public void convertAndSend(final Order<?> order, String receivedSubmissionId)
                 EtorMetadataStep.ETOR_PROCESSING_TAG_ADDED_TO_MESSAGE_HEADER);
 
         String sentSubmissionId = sender.send(omlOrder).orElse(null);
```
Comment (somesylvie) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java:56]: Should this `sentSubmissionId` variable also be some flavor of `reportId`?

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamSenderHelper.java [hunk 2]
```diff
@@ -55,23 +55,23 @@ protected Optional<String> sendToReportStream(
...
+        return outboundReportId;
     }
 
-    protected Optional<String> getSubmissionId(String rsResponseBody) {
+    protected Optional<String> getReportId(String rsResponseBody) {
         try {
             Map<String, Object> rsResponse =
                     formatter.convertJsonToObject(rsResponseBody, new TypeReference<>() {});
-            return Optional.ofNullable(rsResponse.get("submissionId").toString());
+            return Optional.ofNullable(rsResponse.get("reportId").toString());
```
Comment (somesylvie) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamSenderHelper.java:72]: I may be misunderstanding this, but it looks like it's parsing a field from a ReportStream response, and we're now looking for a different field than we were before. Is that the intention/are both `submissionId` and `reportId` real fields and we're switching which we're using?

**Review by somesylvie — COMMENTED** [review]
Are we planning to refactor away other uses of `submissionId`, or are some appropriate to keep? It looks like it's in a bunch of other places in the codebase still

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamSenderHelper.java (hunk 2 above)
Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamSenderHelper.java:72]: Exactly.  Both `submissionId` and `reportId` are fields in a real RS response.  We are changing what we're using.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java (hunk 1 above)
Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java:56]: Good catch.

**Comment by halprin** [comment]
> Are we planning to refactor away other uses of `submissionId`, or are some appropriate to keep? It looks like it's in a bunch of other places in the codebase still

We are, but I'm not sure if this is the PR to do it everywhere.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java (hunk 1 above)
Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java]: Fixed.

**Comment by basiliskus** [comment]
> > Are we planning to refactor away other uses of `submissionId`, or are some appropriate to keep? It looks like it's in a bunch of other places in the codebase still
> 
> We are, but I'm not sure if this is the PR to do it everywhere.

Should we create a devex/opex story to rename these in the rest of the codebase?

**Review by basiliskus — APPROVED** [review]

**Comment by halprin** [comment]
> > > Are we planning to refactor away other uses of `submissionId`, or are some appropriate to keep? It looks like it's in a bunch of other places in the codebase still
> > 
> > 
> > We are, but I'm not sure if this is the PR to do it everywhere.
> 
> Should we create a devex/opex story to rename these in the rest of the codebase?

Definitely.

**Pull request merged**